### PR TITLE
Fix BMS specs

### PR DESCRIPTION
**Pull request by pasitlaothamatas**
### Summary

Fix specs for AIRs and SoC.

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [ ] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [ ] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

File: specs.md [hunk 1]
```diff
@@ -167,16 +167,16 @@ BMS-6 | General voltage and temperature limits | The BMS must throw an AIR shutd
...
-BMS-11 | AIRs weld/stuck open detection | The BMS must check that the AIR+ and AIR- are in the desired open or closed state at 1kHz, and if not the BMS must throw an AIR shutdown fault and enter the fault state.
+BMS-11 | AIRs weld/stuck open detection | The BMS must check that the AIR+ and AIR- are in the desired open or closed state at 100Hz, and if not the BMS must throw an AIR shutdown fault and enter the fault state.
 BMS-36 | IMD data transmission | The BMS must transmit the high/low status of the IMD's OK_HS output, the information encoded in the IMD's PWM output, and the seconds elapsed since the IMD was powered on.
 BMS-37 | OK status transmission | The BMS must transmit the on/off status of BMS_OK, IMD_OK and BSPD_OK at 100Hz or faster.
-BMS-38 | AIR states transmission | The BMS must transmit the open/closed states of the AIR+ and AIR- over CAN at 100Hz or faster.
-
+BMS-38 | AIR states transmission | The BMS must transmit the open/closed states of the AIR+ and AIR- over CAN at 100Hz or faster. The open/closed states of the AIR+ and AIR- is determined by the AIR_POWER_STATUS and AIR_TOTAL_ISENSE.
```
Comment (pasitlaothamatas) [inline specs.md:173]: @kirons96 Not sure if you wanted me to specify the LUT in the specs or not.

Comment (kirons96) [reply specs.md:173]: nope

File: specs.md [hunk 2]
```diff
@@ -167,16 +167,16 @@ BMS-6 | General voltage and temperature limits | The BMS must throw an AIR shutd
...
 BMS-12 | Precharge | - The BMS must wait for 5 seconds after boot, then wait for the closing of the AIR-, to execute the precharge sequence. <br/> - The BMS must precharge the inverter/charger capacitors to at least 98% of the accumulator voltage for extra safety margin. <br/> - Upon a successful precharge, the BMS must close the AIR+. <br/> <br/> Upon a precharge failure, the BMS must throw an AIR shutdown fault. A precharge failure occurs when: <br/> - The TS (tractive system) bus voltage does not rise within the allotted time. <br/> - The TS bus voltage rises too quickly. ([TODO: calculate constants](https://github.com/UBCFormulaElectric/Consolidated-Firmware/issues/515))| EV.6.9.1
-BMS-35 | SoC retrieval | The BMS must retrieve SoC from three different EEPROM regions, and use a voting algorithm to identify which data is correct, in case of data corruption.
+BMS-35 | SoC retrieval | The BMS must retrieve SoC from three different EEPROM regions, and use a voting algorithm to identify which data is correct, in case of data corruption. In the case of data corruption, the BMS must retrieve the SoC from a lookup table generated by the previously gathered discharge data.
```
[the remark this answers is not in the document]
Comment (kirons96) [reply specs.md:179]: this is probably overkill in retrospect, the EEPROM in the MCU should work fine and the voting algorithm makes it extra safe.

instead, why not retry retrieving the data from the EEPROM and if that fails, just set to 50% SoC for the memes?

[the remark this answers is not in the document]
Comment (kirons96) [reply specs.md:179]: i'm saying this because we don't have a LUT and probably will never have time to get one

File: specs.md [hunk 3]
```diff
@@ -167,16 +167,16 @@ BMS-6 | General voltage and temperature limits | The BMS must throw an AIR shutd
 BMS-7 | Charge temperature limits | The BMS must throw an AIR shutdown fault and enter the fault state if charging is attempted outside of these bounds: <br/>  0.0C < any cell temperature < 45.0C. | EV.5.1.3, EV.5.1.10
 BMS-9 | Charger detection and logging | - The BMS must check the charger connection status at 1Hz by the state of the CHARGE_STATE_3V3 digital input. <br/> - The BMS must log the charger connection status over CAN at 1Hz or faster.
 BMS-10 | Charger enable/disable | The BMS must enable the charger by setting the BMS PON pin high and disable the charger by setting the BMS PON pin low.
-BMS-11 | AIRs weld/stuck open detection | The BMS must check that the AIR+ and AIR- are in the desired open or closed state at 1kHz, and if not the BMS must throw an AIR shutdown fault and enter the fault state.
+BMS-11 | AIRs weld/stuck open detection | The BMS must check that the AIR+ and AIR- are in the desired open or closed state at 100Hz, and if not the BMS must throw an AIR shutdown fault and enter the fault state.
```
Comment (kirons96) [inline specs.md:170]: i don't think we can detect whether AIR- is open or closed right?

**Review by kirons96 — CHANGES_REQUESTED** [review]

File: specs.md (hunk 3 above)
Comment (pasitlaothamatas) [reply specs.md:170]: Oh whoops

Comment (pasitlaothamatas) [reply specs.md:170]: Wait I thought from this LUT that we could detect it, but cannot open/close.

image

File: specs.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (pasitlaothamatas) [reply specs.md:179]: > this is probably overkill in retrospect, the EEPROM in the MCU should work fine and the voting algorithm makes it extra safe.
> 
> instead, why not retry retrieving the data from the EEPROM and if that fails, just set to 50% SoC for the memes?

Is this safe?

File: specs.md [hunk 4]
```diff
@@ -218,7 +218,7 @@ BMS-30 | Exiting the fault state and entering the init state | Once all AIR shut
 ID | Title | Description | Associated Competition Rule(s)
 --- | --- | --- | ---
 DIM-0 | Startup CAN message | The DIM must transmit a startup message over CAN on boot.
-DIM-12 | State CAN message | The DCM must transmit the state of its state machine at 100Hz or faster.
+DIM-12 | State CAN message | The DIM must transmit the state of its state machine at 100Hz or faster.
```
Comment (pasitlaothamatas) [inline specs.md:221]: unrelated but im certain this should say DIM and not DCM

File: specs.md [hunk 5]
```diff
@@ -191,7 +191,7 @@ BMS-18 | Cell balancing | - The BMS must balance the cells until they are all be
 BMS-19 | Power limits calculation and sending (charge state) | - The BMS must calculate charge power limits based on cell temperatures and SoC to avoid exceeding a cell's defined limits. <br/> - The BMS must send the charge power limits to the charger over CAN at 100Hz or faster.
 BMS-20 | Charger disconnection | Upon sensing charger disconnection, the BMS must throw an AIR shutdown fault and enter the fault state.
 BMS-21 | Entering the charge state | The BMS must only enter the charge state after the init state is complete.
```
Comment (pasitlaothamatas) [inline specs.md:193]: @kirons96 is this not repeating BMS-16?

File: specs.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kirons96) [reply specs.md:179]: we have bigger problems if we can't read SoC from the internal MCU EEPROM, there's probably an issue with the MCU at that point. i think it's safe

File: specs.md (hunk 5 above)
Comment (kirons96) [reply specs.md:193]: yeah, but i was just explicit about entering and exiting every possible state

File: specs.md (hunk 3 above)
Comment (kirons96) [reply specs.md:170]: i did come up with that truth table... i have forgotten how this car works

Comment (kirons96) [reply specs.md:170]: this is fine then

**Review by kirons96 — CHANGES_REQUESTED** [review]

File: specs.md [hunk 6]
```diff
@@ -167,16 +167,16 @@ BMS-6 | General voltage and temperature limits | The BMS must throw an AIR shutd
...
 ID | Title | Description | Associated Competition Rule(s)
 --- | --- | --- | ---
 BMS-12 | Precharge | - The BMS must wait for 5 seconds after boot, then wait for the closing of the AIR-, to execute the precharge sequence. <br/> - The BMS must precharge the inverter/charger capacitors to at least 98% of the accumulator voltage for extra safety margin. <br/> - Upon a successful precharge, the BMS must close the AIR+. <br/> <br/> Upon a precharge failure, the BMS must throw an AIR shutdown fault. A precharge failure occurs when: <br/> - The TS (tractive system) bus voltage does not rise within the allotted time. <br/> - The TS bus voltage rises too quickly. ([TODO: calculate constants](https://github.com/UBCFormulaElectric/Consolidated-Firmware/issues/515))| EV.6.9.1
-BMS-35 | SoC retrieval | The BMS must retrieve SoC from three different EEPROM regions, and use a voting algorithm to identify which data is correct, in case of data corruption.
+BMS-35 | SoC retrieval | The BMS must retrieve SoC from three different EEPROM regions, and use a voting algorithm to identify which data is correct, in case of data corruption. In the case of data corruption, set the SoC to 50%.
```
Comment (pasitlaothamatas) [inline specs.md:179]: Alright this should be updated now

**Review by kirons96 — APPROVED** [review]

**Pull request merged**